pkg/api/probe.go

```diff
@@ -1,44 +1,80 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
 package api
 
 import "sync/atomic"
 
+// ProbeStatus is the status of a probe.
+// ProbeStatus is treated as a sync/atomic int32.
+type ProbeStatus int32
+
+// get returns the value of the ProbeStatus.
+func (ps *ProbeStatus) get() ProbeStatus {
+	return ProbeStatus(atomic.LoadInt32((*int32)(ps)))
+}
+
+// set updates the value of the ProbeStatus.
+func (ps *ProbeStatus) set(v ProbeStatus) {
+	atomic.StoreInt32((*int32)(ps), int32(v))
+}
+
+// String implements the fmt.Stringer interface.
+func (ps ProbeStatus) String() string {
+	switch ps.get() {
+	case ProbeStatusOK:
+		return "ok"
+	case ProbeStatusNOK:
+		return "nok"
+	}
+	return "unknown"
+}
+
 const (
-	probeStatusPending int32 = iota
-	ProbeStatusOK
-	ProbeStatusNOK
+	// ProbeStatusOK indicates positive ProbeStatus status.
+	ProbeStatusOK ProbeStatus = 1
+
+	// ProbeStatusNOK indicates negative ProbeStatus status.
+	ProbeStatusNOK ProbeStatus = 0
 )
 
-// Probe structure holds flags which indicate node healthiness (k8s equivalent for liveness) and readiness.
+// Probe structure holds flags which indicate node healthiness (sometimes refert also as liveness) and readiness.
 type Probe struct {
-	// Healthy probe indicates node overall health.
-	healthy int32
+	// Healthy probe indicates if node, due to any reason, needs to restarted.
+	healthy ProbeStatus
 	// Ready probe indicates that node is ready to start accepting traffic.
-	ready int32
+	ready ProbeStatus
 }
 
+// NewProbe returns new Probe.
 func NewProbe() *Probe {
```

Comment:
The constructor can be dropped and replaced with `new(Probe)`, since it doesn't add much value.